app/controllers/restaurants_controller.rb

```diff
@@ -0,0 +1,44 @@
+class RestaurantsController < ApplicationController
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+
+  def index
+    @restaurants = Restaurant.all
+
+    render json: @restaurants
+  end
+
+  def show
+    render json: @restaurant
+  end
+
+  def create
+    @restaurant = Restaurant.new(restaurant_params)
+
+    if @restaurant.save
+      render json: @restaurant, status: :created, location: @restaurant
+    else
+      render json: @restaurant.errors, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @restaurant.update(restaurant_params)
+      render json: @restaurant
+    else
+      render json: @restaurant.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @restaurant.destroy
+  end
+
+  private
+    def set_restaurant
+      @restaurant = Restaurant.find(params[:id])
+    end
+
+    def restaurant_params
+      params.permit(:name, :cuisine, :rating, :ten_bis, :address, :max_delivery_time)
```

Comment:
If you want, you can also `require` parameters to ensure they were provided. For example `name` because it can't be null.